src/mlpack/tests/main_tests/gmm_probability_test.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file gmm_probability_test.cpp
+ * @author Yashwant Singh
+ *
+ * Test mlpackMain() of gmm_probability_main.cpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+
+static const std::string testName = "GmmProbability";
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/methods/gmm/gmm_probability_main.cpp>
+
+#include "test_helper.hpp"
+
+#include <boost/test/unit_test.hpp>
+
+
+using namespace mlpack;
+
+struct GmmProbabilityTestFixture
+{
+ public:
+  GmmProbabilityTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+
+  ~GmmProbabilityTestFixture()
+  {
+    // Clear the settings.
+    CLI::ClearSettings();
+  }
+};
+
+void ResetGmmProbabilitySetting()
+{
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+}
+
+BOOST_FIXTURE_TEST_SUITE(GmmProbabilityMainTest, GmmProbabilityTestFixture);
+
+// Making sure input_file are provided.
+BOOST_AUTO_TEST_CASE(GmmProbabilityInputTest)
+{
+  arma::mat inputData(5, 10, arma::fill::randu);
+
+  GMM gmm(1, 5);
+  gmm.Train(inputData, 5);
+
+  SetInputParam("input_model", &gmm);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
```

Comment:
So, for that reason I described in the other comment, we don't need to include this test---we already know a user will have to pass both `input_model` and `input`.

Reply:
I have remove test as suggested.